client-misk-hibernate/src/main/kotlin/app/cash/backfila/client/misk/hibernate/IdPrimaryKeyCursorAdapter.kt

```diff
@@ -0,0 +1,18 @@
+package app.cash.backfila.client.misk.hibernate
+
+import misk.hibernate.DbEntity
+import misk.hibernate.Id
+import okio.ByteString
+import okio.ByteString.Companion.encodeUtf8
+
+internal object IdPrimaryKeyCursorAdapter : PrimaryKeyCursorAdapter<Id<*>> {
+  override fun toByteString(pk: Id<*>): ByteString {
+    return pk.toString().encodeUtf8()
+  }
+
+  override fun fromByteString(byteString: ByteString): Result<Id<*>> {
```

Comment:
why Result: It forces you to check, and gives an obvious way to fail

Reply:
I like it.